src/metaxy/metadata_store/base.py

```diff
@@ -515,34 +567,77 @@ def write_metadata(
                     UserWarning,
                     stacklevel=2,
                 )
-        else:
-            # Get current feature version and snapshot_version from code and add them
-            if isinstance(feature, type) and issubclass(feature, BaseFeature):
-                current_feature_version = feature.feature_version()  # type: ignore[attr-defined]
-            else:
-                from metaxy.models.feature import FeatureGraph
-
-                graph = FeatureGraph.get_active()
-                feature_cls = graph.features_by_key[feature_key]
-                current_feature_version = feature_cls.feature_version()  # type: ignore[attr-defined]
+            return df
 
-            # Get snapshot_version from active graph
+        # Get current feature version
+        if isinstance(feature, type) and issubclass(feature, BaseFeature):
+            current_feature_version = feature.feature_version()  # type: ignore[attr-defined]
+        else:
             from metaxy.models.feature import FeatureGraph
 
             graph = FeatureGraph.get_active()
-            current_snapshot_version = graph.snapshot_version
+            feature_cls = graph.features_by_key[feature_key]
```

Comment:
I am pretty sure there already is a helper method that does this conversion, let's reuse it

Reply:
done